stk/utilities/utilities.py

```diff
@@ -1143,10 +1146,450 @@ def vector_theta(vector1, vector2):
     """
 
     numerator = np.dot(vector1, vector2)
-    denominator = (np.linalg.norm(vector1) *
-                   np.linalg.norm(vector2))
+    denominator = (np.linalg.norm(vector1) * np.linalg.norm(vector2))
     # This if statement prevents returns of NaN due to floating point
     # incurracy.
     if np.isclose(numerator, denominator, atol=1e-8):
         return 0.0
     return np.arccos(numerator/denominator)
+
+
+class XTBInvalidSolventError(Exception):
+    ...
+
+
+def is_valid_xtb_solvent(gfn_version, solvent):
+    """
+    Check if solvent is valid for the given GFN version.
+
+    Parameters
+    ----------
+    gfn_version : :class:`int`
+        GFN parameterization version. Can be: ``0``, ``1`` or ``2``.
+
+    solvent : :class:`str`
+        Solvent being tested.
+
+    Returns
+    -------
+    :class:`bool`
+        ``True`` if solvent is valid.
+
+    See Also
+    --------
+    # https://xtb-docs.readthedocs.io/en/latest/gbsa.html
+
+    """
+    if gfn_version == 0:
+        return False
+    elif gfn_version == 1:
+        valid_solvents = {
+            'acetone', 'acetonitrile', 'benzene',
+            'CH2Cl2'.lower(), 'CHCl3'.lower(), 'CS2'.lower(),
+            'DMSO'.lower(), 'ether', 'H2O'.lower(),
+            'methanol', 'THF'.lower(), 'toluene'
+        }
+        return solvent in valid_solvents
+    elif gfn_version == 2:
+        valid_solvents = {
+            'acetone', 'acetonitrile', 'CH2Cl2'.lower(),
+            'CHCl3'.lower(), 'CS2'.lower(), 'DMF'.lower(),
+            'DMSO'.lower(), 'ether', 'H2O'.lower(), 'methanol',
+            'n-hexane'.lower(), 'THF'.lower(), 'toluene'
+        }
+        return solvent in valid_solvents
+
+
+class XTBExtractor:
+    """
+    Extracts properties from GFN-xTB output files.
+
+    Attributes
+    ----------
+    output_file : :class:`str`
+        Output file to extract properties from.
+
+    output_list : :class:`list`
+        :class:`list` of all lines in as :class:`str` in output file.
+
```

Comment:
This class is missing the documentation for all the extracted attributes.
once those are documented here, you can remove the things like 

```
        Gives components of quadrupole moment as
        ``[x, y, z]`` in a :class:`list`.

```

in the extract method's documentation. Instead you can just write

```

Updates :attr:`qonly_quadrupole_moment`.

```

Which will prevent the need to document the same thing twice in in different places (which leads to a high chance that one  of them will forget to be updated when the other changes)